nixos/modules/services/continuous-integration/gitlab-runner.nix

```diff
@@ -66,6 +66,12 @@ let
             ++ optional service.debugTraceDisabled
             "--debug-trace-disabled"
             ++ map (e: "--env ${escapeShellArg e}") (mapAttrsToList (name: value: "${name}=${value}") service.environmentVariables)
+            ++ optionals (service.executor == "docker+machine") (
+              assert (
+                assertMsg (service.dockerImage != null)
+                  "dockerImage option is required for docker+machine executor (${name})");
+              [ "--docker-image ${service.dockerImage}" ]
+            )
```

Comment:
Are other `--docker-` options not used with `docker+machine`? How about `docker-ssh+machine`?

Reply:
Hmm, that is a good point. I actually do not know, since I am not (currently) using them.

I was surprised by the runtime error (PANIC: …), so I wanted to add the assert for docker+machine at least.

Reply:
How is `gitlab-runner register` reacting to extraneous args? Maybe the `--docker-*` options do not need to be under the check?

Reply:
Tested https://github.com/NixOS/nixpkgs/pull/108763 `docker+machine` and `docker-ssh+machine` (`--ssh-*` flags required) with `dockerVolumes` set - no errors.